02_combine_counts/combine_counts/combine_counts.py

```diff
@@ -0,0 +1,52 @@
+import argparse
+import os
+import sys
+
+import pandas as pd
+
+
+def merge_counts(count_files):
+    base_names = [os.path.basename(file_name) for file_name in count_files]
+    return [
+        pd.read_csv(
+            file_name,
+            sep="\t",
+            index_col=0,
+            names=[
+                base_name.replace("_counts.txt", ""),
+                base_name.replace("ref_counts.txt", "alt"),
+            ],
+        )
+        for file_name, base_name in zip(count_files, base_names)
+    ]
+
+
+def run(args):
+    inputs = sorted(args.inputs)
+    outputs = sorted(args.outputs)
+    input_df = pd.concat(merge_counts(inputs), axis=1).sort_index()
+    output_df = pd.concat(merge_counts(outputs), axis=1).sort_index()
+
+    input_df.insert(0, "input_count", len(inputs))
+    output_df.insert(0, "output_count", len(outputs))
+    df = pd.concat([input_df, output_df], axis=1).fillna(0).astype(int)
```

Comment:
This seems a bit cluttered to me, with an additional custom merge function instead of using pandas joins. What is the advantage of replacing the looped `join`s for these `concat`s? Where you able to reproduce existing files with this code?

Reply:
I've changed the code so it's simpler -- still uses concat though, which I think is semantically the right tool here.

Reply:
Are all variants guaranteed to be in all replicates? I don't think so. And if that is not the case, using `pd.concat` in lines 27 and 28 is wrong (hence the need to join over the variant ids).

Reply:
I'm pretty sure concat does the right thing when combing two dataframes with indexes that don't 100% match:
```
In [11]: a = pd.DataFrame([[1, 2], [3, 4]], index=["a", "b"])

In [12]: b = pd.DataFrame([[5, 6], [7, 8]], index=["a", "c"])

In [15]: pd.concat([a, b], axis=1).fillna(0).astype(int)
Out[15]:
   0  1  0  1
a  1  2  5  6
b  3  4  0  0
c  0  0  7  8
```